deployer/src/persistence/mod.rs

```diff
@@ -296,6 +296,24 @@ impl Persistence {
         .map_err(Error::from)
     }
 
+    /// Gets a deployment if it is runnable
+    pub async fn get_runnable_deployment(&self, id: &Uuid) -> Result<Option<DeploymentRunnable>> {
+        sqlx::query_as(
+            r#"SELECT d.id, service_id, s.name AS service_name, d.is_next
+                FROM deployments AS d
+                JOIN services AS s ON s.id = d.service_id
+                WHERE state IN (?, ?, ?)
+                AND d.id = ?"#,
+        )
+        .bind(State::Running)
+        .bind(State::Stopped)
+        .bind(State::Completed)
```

Comment:
nit: the `Completed` deployments might be stateful and be prevented from running again (like a `FnOnce`?), but they can also be stateless. Not sure if it is worth including them here, but can't see them as a big problem (they would also be restarted only if they are idled for some reason, but that would be conflicting with the fact that they are completed, just because that means they should exit by themselves and the project state should be `ProjectStopped` probably?).

Reply:
Hmm, interesting. :thinking:

Reply:
I think not covering `Complete` means users won't be able to start stateless projects ever.

But covering `Complete` means users can use some idempotency to get around it. So this seems to cover more options in the future :smile: